### PR TITLE
[EDA] Settings Saving Functionality

### PR DESCRIPTION
**Pull request by m-kinkade**
# Critical Changes

# Changes

# Issues Closed

# New Metadata
### Custom Labels
stgErrorInsufficientAccessSaving
stgErrorInvalidSettings

### Apex Classes
HierarchySettingsChangesController
HierarchySettingsChangesController_TEST
HierarchySettingsService
HierarchySettingsService_TEST
HierarchySettingsMapper
HierarchySettingsMapper_TEST
HierarchySettingsChangesModel
HierarchySettingsChangesModel_TEST
HierarchySettingsModel
HierarchySettingsModel_TEST

## Unpackaged Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: force-app/main/default/classes/HierarchySettingsChangesModel.cls
Comment (MatthewBlanski) [inline force-app/main/default/classes/HierarchySettingsChangesModel.cls:46]: This extra line feels odd here.

File: force-app/main/default/classes/HierarchySettingsChangesModel.cls
Comment (MatthewBlanski) [inline force-app/main/default/classes/HierarchySettingsChangesModel.cls:36]: These need to be aura enabled

File: force-app/main/default/classes/HierarchySettingsChangesModel.cls
Comment (MatthewBlanski) [inline force-app/main/default/classes/HierarchySettingsChangesModel.cls:34]: Now that we are pulling it down, we probably should have this be a view model instead.
```suggestion
public with sharing class HierarchySettingsChangesVModel {
```

File: force-app/main/default/classes/HierarchySettingsService.cls
Comment (MatthewBlanski) [inline force-app/main/default/classes/HierarchySettingsService.cls:81]: This is still commented out. Are we still waiting on that service? We can merge in the other branch and define this as a dependency for it.

File: force-app/main/default/classes/HierarchySettingsService_TEST.cls
Comment (MatthewBlanski) [inline force-app/main/default/classes/HierarchySettingsService_TEST.cls:113]: Likewise here for the commenting

File: force-app/main/default/classes/HierarchySettingsMapper.cls
Comment (MatthewBlanski) [inline force-app/main/default/classes/HierarchySettingsMapper.cls:140]: This method should be moved into the service.

File: force-app/main/default/classes/HierarchySettingsMapper.cls
Comment (MatthewBlanski) [inline force-app/main/default/classes/HierarchySettingsMapper.cls:150]: This method should be moved into the service.

**Review by MatthewBlanski — CHANGES_REQUESTED** [review]
A few changes requested.

**Comment by m-kinkade** [comment]
This branch has dependencies on the UserPermissionsService.  Merge this branch AFTER https://github.com/SalesforceFoundation/EDA/pull/1581

File: force-app/main/default/lwc/settingsSaveCanvas/settingsSaveCanvas.js
Comment (MatthewBlanski) [reply force-app/main/default/lwc/settingsSaveCanvas/settingsSaveCanvas.js:83]: Let's excise it here @m-kinkade

**Review by MatthewBlanski — APPROVED** [review]
This looks like it's working! I'm so excited.

**Comment by spinkelman** [comment]
Limited QA passes - Checked over code and labels, but further testing needed once functioning fields are added.

**Pull request merged**
